File: app/src/main/java/com/ntsoftware/vspc/myvspc/screens/discover/DiscoverAdapter.java
package com.ntsoftware.vspc.myvspc.screens.discover;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.ntsoftware.vspc.myvspc.R;
import  com.ntsoftware.vspc.myvspc.screens.discover.holders.HolderType;
import com.ntsoftware.vspc.myvspc.screens.discover.holders.ScheduleHolder;
import com.ntsoftware.vspc.myvspc.screens.discover.holders.TodayHolder;
import com.ntsoftware.vspc.myvspc.screens.discover.holders.WeatherHolder;
import com.ntsoftware.vspc.myvspc.screens.discover.holders.base.DiscoverHolder;
import com.ntsoftware.vspc.myvspc.screens.discover.holders.base.EmtyHolder;

public class DiscoverAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    public DiscoverAdapter() {
    }

    @Override
    public int getItemViewType(int position) {
        if (position == HolderType.TODAY.ordinal()) {
            return HolderType.TODAY.ordinal();
        }
        else if (position == HolderType.WEATHER.ordinal()) {
            return HolderType.WEATHER.ordinal();
        }
        else if (position == HolderType.SCHEDULE.ordinal()) {
            return HolderType.SCHEDULE.ordinal();
        }
        else if (position == HolderType.SIMPLE_NEWS.ordinal()) {
            return HolderType.SIMPLE_NEWS.ordinal();
        }
        else return 0;
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        RecyclerView.ViewHolder viewHolder = null;

        View holderLayoutView;

        if (viewType == HolderType.TODAY.ordinal()) {
            holderLayoutView = LayoutInflater.from(parent.getContext()).inflate(R.layout.rv_item_discover_today,parent,false);
            viewHolder = new TodayHolder(holderLayoutView);
        }
        else if (viewType == HolderType.WEATHER.ordinal()) {
            holderLayoutView = LayoutInflater.from(parent.getContext()).inflate(R.layout.rv_item_discover_weather,parent,false);
            viewHolder = new WeatherHolder(holderLayoutView);
        }
        else if (viewType == HolderType.SCHEDULE.ordinal()) {
            holderLayoutView = LayoutInflater.from(parent.getContext()).inflate(R.layout.rv_item_discover_schedule,parent,false);
            viewHolder = new ScheduleHolder(holderLayoutView);
        }
//        else if (viewType == HolderType.SIMPLE_NEWS.ordinal()) {
//
//        }
        else {
            holderLayoutView = LayoutInflater.from(parent.getContext()).inflate(R.layout.rv_item_discover_empty,parent,false);
            viewHolder = new EmtyHolder(holderLayoutView);
        }
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
        ((DiscoverHolder)holder).bind();

    }

    @Override
    public int getItemCount() {
        return 4;
    }
}

File: app/src/main/java/com/ntsoftware/vspc/myvspc/screens/discover/DiscoverAdapter.kt
package com.ntsoftware.vspc.myvspc.screens.discover

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.ntsoftware.vspc.myvspc.R
import com.ntsoftware.vspc.myvspc.screens.discover.holders.HolderType
import com.ntsoftware.vspc.myvspc.screens.discover.holders.ScheduleHolder
import com.ntsoftware.vspc.myvspc.screens.discover.holders.TodayHolder
import com.ntsoftware.vspc.myvspc.screens.discover.holders.WeatherHolder
import com.ntsoftware.vspc.myvspc.screens.discover.holders.base.DiscoverHolder
import com.ntsoftware.vspc.myvspc.screens.discover.holders.base.EmptyHolder

class DiscoverAdapter: RecyclerView.Adapter<RecyclerView.ViewHolder>() {

    override fun getItemViewType(position: Int): Int {
        when(position) {
            HolderType.TODAY.ordinal -> return HolderType.TODAY.ordinal
            HolderType.WEATHER.ordinal -> return HolderType.WEATHER.ordinal
            HolderType.SCHEDULE.ordinal -> return HolderType.SCHEDULE.ordinal
            HolderType.SIMPLE_NEWS.ordinal -> return HolderType.SIMPLE_NEWS.ordinal
            else -> return 0
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        var holder: RecyclerView.ViewHolder? = null

        var holder_layout: View

        when(viewType){
            HolderType.TODAY.ordinal -> {
                holder_layout = LayoutInflater.from(parent.context).inflate(R.layout.rv_item_discover_today,parent,false)
                holder = TodayHolder(holder_layout)
            }
            HolderType.WEATHER.ordinal -> {
                holder_layout = LayoutInflater.from(parent.context).inflate(R.layout.rv_item_discover_weather,parent,false)
                holder = WeatherHolder(holder_layout)
            }
            HolderType.SCHEDULE.ordinal -> {
                holder_layout = LayoutInflater.from(parent.context).inflate(R.layout.rv_item_discover_schedule,parent,false)
                holder = ScheduleHolder(holder_layout)
            }
//            HolderType.SIMPLE_NEWS.ordinal -> {
//
//            }
            else -> {
                holder_layout = LayoutInflater.from(parent.context).inflate(R.layout.rv_item_discover_empty,parent,false)
                holder = EmptyHolder(holder_layout)
            }
        }

        return holder
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        (holder as DiscoverHolder).bind()
    }

    override fun getItemCount(): Int {
        return 4;
    }
}